pkg/kubelet/cm/dra/plugin/dra_plugin_test.go

```diff
@@ -374,3 +373,108 @@ func TestPlugin_WatchResources(t *testing.T) {
+func TestUnifiedConnectionUsage(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	addr := path.Join(t.TempDir(), "dra.sock")
+	teardown, err := setupFakeGRPCServer("", addr) // All services registered
+	require.NoError(t, err)
+	defer teardown()
+
+	driverName := "test-driver"
+	draPlugins := NewDRAPluginManager(tCtx, nil, nil, &mockStreamHandler{}, 0)
+	err = draPlugins.add(driverName, addr, drapbv1.DRAPluginService, defaultClientCallTimeout)
+	require.NoError(t, err)
+	defer draPlugins.remove(driverName, addr)
+
+	plugin, err := draPlugins.GetPlugin(driverName)
+	require.NoError(t, err)
+
+	// Perform DRA operation first
+	req := &drapbv1.NodePrepareResourcesRequest{
+		Claims: []*drapbv1.Claim{
+			{
+				Namespace: "test-namespace",
+				Uid:       "test-uid",
+				Name:      "test-claim",
+			},
+		},
+	}
+	_, err = plugin.NodePrepareResources(tCtx, req)
+	require.NoError(t, err)
+
+	// Capture connection reference after DRA operation
+	firstConn := plugin.conn
+	require.NotNil(t, firstConn, "Connection should exist after DRA operation")
+
+	// Now perform Health operation - should use same connection
+	ctx, cancel := context.WithTimeout(tCtx, 5*time.Second)
+	defer cancel()
+
+	stream, err := plugin.NodeWatchResources(ctx)
+	require.NoError(t, err)
+	require.NotNil(t, stream)
+
+	// Verify same connection is used
+	require.Equal(t, firstConn, plugin.conn, "Health operation should use same connection as DRA operation")
+
+	// Verify we can still receive health data
+	resp, err := stream.Recv()
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	require.Len(t, resp.Devices, 1)
+}
+
+func TestConnectionDrop(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	addr := path.Join(t.TempDir(), "dra.sock")
+
+	teardown, err := setupFakeGRPCServer("", addr)
+	require.NoError(t, err)
+
+	driverName := "test-driver"
+	draPlugins := NewDRAPluginManager(tCtx, nil, nil, &mockStreamHandler{}, 0)
+	err = draPlugins.add(driverName, addr, drapbv1.DRAPluginService, defaultClientCallTimeout)
+	require.NoError(t, err)
+	defer draPlugins.remove(driverName, addr)
+
+	plugin, err := draPlugins.GetPlugin(driverName)
+	require.NoError(t, err)
+
+	// Verify DRA operations still work
+	req := &drapbv1.NodePrepareResourcesRequest{
+		Claims: []*drapbv1.Claim{{
+			Namespace: "test-ns",
+			Uid:       "test-uid",
+			Name:      "test-claim",
+		}},
+	}
+	_, err = plugin.NodePrepareResources(tCtx, req)
+	require.NoError(t, err, "DRA operations should still work")
+
+	// Verify health code works and uses same connection
+	ctx, cancel := context.WithTimeout(tCtx, 2*time.Second)
+	defer cancel()
+
+	stream, err := plugin.NodeWatchResources(ctx)
+	require.NoError(t, err, "Health stream should work")
+
+	resp, err := stream.Recv()
+	require.NoError(t, err, "Should receive health data")
+	require.Len(t, resp.Devices, 1)
+
+	// Test connection drop handling
+	teardown() // Stop server
+	time.Sleep(50 * time.Millisecond)
```

Comment:
Why it's 50 millisecond? Would this test work reliable on a slower machine? Can we avoid sleeping and test some condition (server stopped?) instead?